operator/api/v1alpha1/httpscaledobject_types.go

```diff
@@ -113,18 +113,23 @@ type ReplicaStruct struct {
 
 // HTTPScaledObjectSpec defines the desired state of HTTPScaledObject
 type HTTPScaledObjectSpec struct {
-	// (optional) The name of the application to be created.
-	AppName string `json:"app_name,omitempty"`
-	// The image this application will use.
-	Image string `json:"app_image"`
-	// The port this application will serve on.
-	Port int32 `json:"port"`
+	// The name of the deployment to route HTTP requests to (and to autoscale). Either this
+	// or Image must be set
+	ScaleTargetRef *ScaleTargetRef `json:"scaleTargetRef"`
 	// (optional) Replica information
 	//+optional
 	Replicas ReplicaStruct `json:"replicas,omitempty"`
 }
 
-// TODO: Add ingress configurations
+// ScaleTargetRef contains all the details about an HTTP application to scale and route to
+type ScaleTargetRef struct {
+	// The name of the deployment to scale according to HTTP traffic
+	Deployment string `json:"deployment"`
```

Comment:
Just a nit. Do you want to target just `Deployments`, what about `StatefulSet` or `CustomResources` with `/scale` subresoure? Might be worth checking how is this handled in ScaledObject: https://github.com/kedacore/keda/blob/6dc4c3524ba6278276cd1338cc7e4e9a7dd88963/api/v1alpha1/scaledobject_types.go#L75-L81

Reply:
I would like to target `Deployment`s for now, and open it up to other HTTP workloads that folks might want to scale in the future. Does that sound ok to you @zroubalik ?

Reply:
@zroubalik I am going to merge this PR now, and we can create a follow-up if you think we need to support `StatefulSet` or other resources with a `/scale` subresource right now.